google/resource_compute_network_endpoint.go

```diff
@@ -65,7 +65,7 @@ range).`,
 			},
 			"port": {
 				Type:        schema.TypeInt,
-				Required:    true,
+				Optional:    true,
```

Comment:
From the docs it looks like "If not specified, the defaultPort for the network endpoint group will be used." Do you know if this means that this will be nil (but fall back to the network endpoint group's value) or if it will be set on creation to whatever the default is? In the latter case this would need to be computed as well as optional.

Reply:
I've try this changes to and I think any API bug in google, so as documentation if we set the Network Endpoint type as "GCE_VM_IP" we don't need port and will use default_port variable which set in Network Endpoint Group.

But In console, port input is required (Different with the docs). 

The another thing when we implement the changes (modified provider) to infrastructure, the provider will set port to 0 and the network endpoint still created (it's diff with the spec of port which must greater than zero in console). 
So we can conclude that GCP not filter the API when creation by API and have different behaviour with docs. The proven is when we want to destroy the resource via terraform, it will turns to error regarding zero port.


Here's the docs I refer to https://cloud.google.com/load-balancing/docs/negs/setting-up-zonal-negs

Reply:
Okay... I _think_ this probably needs to be computed as well as optional.

Reply:
If you're upstreaming this to magic-modules, you would want to use `default_from_api: true` to indicate that it should be optional & computed. I believe that marking it as computed will probably / hopefully resolve the issue you're seeing with a 0 port? It could also be a problem with Terraform not being able to tell the difference between nil and zero.